### PR TITLE
AppScan Enterprise Project Connector

### PR DESCRIPTION
**Pull request by geoffwhittington**

### CODE REVIEW

**Comment by geoffwhittington** [comment]
@ehsanf I whipped up a simple AppScan Enterprise project connector. I'll see if we can get it merged into master next sprint sometime, if you're interested.

File: sdetools/sde_plugins/connector_plugin/examples/appscan.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
+import json
```
Comment (hanifvirani) [inline sdetools/sde_plugins/connector_plugin/examples/appscan.py]: Alphabetic sorting and blank line between 3rd party and our stuff.

File: sdetools/sde_plugins/connector_plugin/examples/appscan.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+            CustomAttribute(name='businessimpact'),
+            CustomAttribute(name='riskrating'),
+            CustomAttribute(name='appscanID')
+        ]
+
+    def read_only_attributes(self):
+        return []
+
+    def fetch_project_data(self, configuration, lookup_data):
+
```
Comment (hanifvirani) [inline sdetools/sde_plugins/connector_plugin/examples/appscan.py]: Delete blank line here.

File: sdetools/sde_plugins/connector_plugin/examples/appscan.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+        opener = urllib2.build_opener(urllib2.HTTPCookieProcessor(cj))
+        headers = {'Content-Type': 'application/json', 'Accept': 'application/json'}
+
+        req = urllib2.Request(req_url, headers=headers, data=json.dumps(credentials))
+
+        try:
+             result = opener.open(req).read()
+        except urllib2.HTTPError, err:
+            if err.code == 401:
+                error_message = 'Cannot authenticate to AppScan Enterprise server. Make sure credentials are correct.'
```
Comment (hanifvirani) [inline sdetools/sde_plugins/connector_plugin/examples/appscan.py]: Could think about extracting all the error messages to an inner Message class like we do in SDE.

**Comment by hanifvirani** [comment]
Should this live under examples?

**Comment by geoffwhittington** [comment]
Should this live under examples? I currently is - are you suggesting it should live somewhere else? I don't know where else, to be honest.

**Comment by hanifvirani** [comment]
Yeah. I mean is it an example implementation? Is this client specific?

**Comment by geoffwhittington** [comment]
This is a generic implementation - not for any client - we'll probably use it for showcasing - but people are free to use it if they prefer. It will likely be copied somewhere else and customized.

**Pull request merged**